### DISCUSSION TITLE
Lokinet AppImage

### DISCUSSION THREAD

**Discussion by Zardon703** [oxen-io/lokinet#2198] [Ideas]
Been trying to install lokinet for parrot os but to no avail. Found lots of advice from other distro users but none seem to work.
(side note: I'm new to linux but been in tech for a couple of years and have "officially" started my cybersecurity journey)
So what I was suggesting is adding an AppImage for new users (such as myself) to get Lokinet up and running and i'll make a YT/odessy vid to explain to newcomers how to run said AppImage.

Thx for reading and take care. '-')/

**Comment by majestrate** [comment]
a full system lokinet as an appimage without being a security hazard would be dependent on the system layer which defeats the purpose of an appimage, flatpak as well but in a disjointedly similar way.

**Comment by Zardon703** [comment]
> a full system lokinet as an appimage without being a security hazard would be dependent on the system layer which defeats the purpose of an appimage, flatpak as well but in a disjointedly similar way.

So, i'm guessing that's a no

**Reply by majestrate** [reply]
it's possible just not practical, given you want it done in a nonhacky and complete way.

**Reply by Zardon703** [reply]
It can be hacky tbh, cause my expectations of how good software works is pretty low

**Reply by majestrate** [reply]
you'll likely want to use tor in that case.

**Reply by Zardon703** [reply]
Just wanted to route my traffic through a dark-net on which I could game and so on (which isn`t practical on tor ping wise)
And not really feeling a VPN either

**Reply by majestrate** [reply]
keep in mind despite the marketing efforts of the snakeoil VPN tunnel providers, VPN software does not mandate all traffic go over the tunnel. full route tunneling is a default setting when you are giving a configuration to users of a vpn proxy service, but for every other VPN tunnel use case, it's not a requirement as they are primarily used to access a private network via a virtual network interface. lokinet is made for both modes of operation and currently shines the brightest with not using a full routing table (exit market coming soon ™️ ).

**Reply by Zardon703** [reply]
Interesting......
So i'm guessing the OXEN team adding an AppImage for lokinet is a no-go? Even on a hardened distro like parrot?

**Reply by jagerman** [reply]
It's not really about the security; the main obstacle with AppImage for Lokinet is that AppImage is designed as essentially a sandbox, where the code and libraries and resources in the sandbox are deliberately isolated from the rest of the system.  It's really meant for user-space programs to be kept separate from the rest of the system, and it's fine as long as you have an application that doesn't need any special system access (which covers a lot of applications), and doesn't want to worry about playing nice with system dependencies.

For lokinet, however, lokinet needs to interact with the system: it needs to create a network tunnel and adjust the system-wide routing rules so that traffic can be routed from any other applications on your system to the lokinet via the tunnel so that you can just turn lokinet on and then use whatever programs you want, with no configuration, in order to have lokinet access.  (You can even go beyond that and put Lokinet on your network's router, if you want and have a sufficiently advance router), to make your entire network traffic go through Lokinet.

This is where AppImage (or other common container formats) breaks down: a persistent, system-level service is something AppImage is explicitly designed not to allow.  (This isn't really specific to AppImage: a Windows portable executable version of Lokinet, or a non-extension, non-root app on macOS would be impossible for Lokinet for essentially the same reason).

**Reply by Zardon703** [reply]
Well that makes perfect sense. Thx for the in depth explanation from both of you.
One final question, y`all know the Tor GUI install prosses right? Why not do something like that instead?
Just a thought
